### PR TITLE
Added Cassandra chart and container

### PR DESCRIPTION
**Pull request by algchoo**
* Added Cassandra container w/ no authentication for remote JMX
* Added chart to create a Cassandra cluster using a StatefulSet and Service

### CODE REVIEW

File: container/cassandra/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+version: "3.9"
+services:
+  cassandra1:
+    restart: always
+    build:
+      context: .
+    ports:
```
Comment (schmikei) [inline container/cassandra/docker-compose.yaml:7]: Does this one need to seed `cassandra1` as well? Or does the default of localhost work in this context?

File: charts/cassandra/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      - name: cassandra
+        image: {{ .Values.image }}
+        env:
+          - name: CASSANDRA_SEEDS
+            value: "cassandra-0.cassandra.default.svc.cluster.local"
+          - name: JVM_OPTS
+            value: "-Xms1024M -Xmx2048M -XX:MaxPermSize=2048M -XX:MaxHeapSize=2048M"
+        resources:
+          limits:
+            cpu: "500m"
```
Comment (jsirianni) [inline charts/cassandra/templates/deployment.yaml:43]: I think it is okay to remove the cpu limit to allow the containers to "burst".

File: charts/cassandra/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        app.kubernetes.io/app: cassandra
+    spec:
+      containers:
+      - name: cassandra
+        image: {{ .Values.image }}
+        env:
+          - name: CASSANDRA_SEEDS
+            value: "cassandra-0.cassandra.default.svc.cluster.local"
+          - name: JVM_OPTS
+            value: "-Xms1024M -Xmx2048M -XX:MaxPermSize=2048M -XX:MaxHeapSize=2048M"
```
Comment (jsirianni) [inline charts/cassandra/templates/deployment.yaml:40]: I would set min and max equal, and I think 1024 would be ideal to allow for any filesystem caching that might be needed on top of heap allocation.

**Review by jsirianni — CHANGES_REQUESTED** [review]
I think we need a couple tweaks and it will be good to go. This is what is working great for me:

```yaml
          - name: JVM_OPTS
            value: "-Xms1024M -Xmx1024M -XX:MaxPermSize=1024M -XX:MaxHeapSize=1024M"
        resources:
          limits:
            memory: 2G
          requests:
            cpu: "500m"
            memory: 2G

```

File: charts/cassandra/templates/deployment.yaml (hunk 3 above)
Comment (jsirianni) [reply charts/cassandra/templates/deployment.yaml]: @algchoo I think we still need this

Comment (jsirianni) [reply charts/cassandra/templates/deployment.yaml]: To clarify, I think heap min and max can be 1GB, with cgroup limit / request set to 2gb. This allows for any filesystem caching / memory requirements required outside of the JVM heap.

**Review by jsirianni — APPROVED** [review]

**Review by schmikei — APPROVED** [review]

**Pull request merged**